### PR TITLE
(GH-1496) Enable ability to create a trimmed down version of Chocolatey.Lib

### LINKED ISSUE #1496: Need ability to have a LibOnly version of Chocolatey.Lib
Currently, the Chocolatey.Lib that is generated as part of the overall Chocolatey build, generates an assembly that contains everything that is needed to run choco.exe.  In some cases, all the contents of this assembly are not required, for example when only running one or two of the choco commands.  In these circumstances, it makes sense to make a trimmed down version of Chocolatey.Lib that only contains the bare minimum of requirements to function.

Obviously, this will mean that this trimmed down version of Chocolatey.Lib should only be used in a few special cases, as it will not have all the functionality of the main Chocolatey.Lib.

### PR DESCRIPTION
**Pull request by gep13**
This includes:

* New Compilation Directive, `LibOnly` to control when certain pieces of code are compiled
* Conditional inclusion of referenced projects
* Modification to build process to only ilmerge certain projects, based on configuration

Fixes #1496

### CODE REVIEW

File: src/chocolatey/GetChocolatey.cs [hunk 1]
```diff
@@ -500,7 +502,9 @@ private void extract_resources()
 
             try
             {
+#if !LibOnly
                 AssemblyFileExtractor.extract_all_resources_to_relative_directory(_container.GetInstance<IFileSystem>(), Assembly.GetAssembly(typeof(ChocolateyResourcesAssembly)), ApplicationParameters.InstallLocation, folders, ApplicationParameters.ChocolateyFileResources);
+#endif
```
Comment (ferventcoder) [inline src/chocolatey/GetChocolatey.cs]: Would it be better to surround the entire try/catch with `#if !LibOnly`? I'm not sure how it likes an empty try with a filled out catch.

**Comment by gep13** [comment]
@ferventcoder fixed up as requested.

**Comment by ferventcoder** [comment]
Looks like this will need rebased.

**Comment by gep13** [comment]
@ferventcoder rebased 😄

**Comment by gep13** [comment]
Okay, looks like I messed up something with the merge...  Give me a couple minutes...

**Comment by gep13** [comment]
@ferventcoder ah, there we go, all better 😸

**Pull request merged**